src/sensor.cpp

```diff
@@ -1158,32 +944,536 @@ namespace librealsense
+            {
+                auto&& target = to_profile(req.get());
+                auto&& mapped_source_profiles = _target_to_source_profiles_map[target];
+
+                for (auto&& source_profile : mapped_source_profiles)
+                {
+                    if (best_pbf->has_source(source_profile))
+                    {
+                        resolved_req_set.insert(source_profile);
+                        current_resolved_reqs.insert(source_profile);
+
+                        // Do not override already defined processing blocks
+                        const auto&& pb = _profiles_to_processing_block.find(source_profile);
+                        if (pb == end(_profiles_to_processing_block))
+                            _profiles_to_processing_block[source_profile] = best_pb;
+                    }
+                }
+            }
+            const stream_profiles&& print_current_resolved_reqs = { current_resolved_reqs.begin(), current_resolved_reqs.end() };
+            LOG_DEBUG("Request: " << best_reqs << "\nResolved to: " << print_current_resolved_reqs);
+        }
+        resolved_req = { resolved_req_set.begin(), resolved_req_set.end() };
+        return resolved_req;
+    }
+
+    void synthetic_sensor::open(const stream_profiles& requests)
+    {
+        std::lock_guard<std::mutex> lock(_synthetic_configure_lock);
+        add_source_profiles_missing_data();
+
+        const auto&& resolved_req = resolve_requests(requests);
+
+        _raw_sensor->set_source_owner(this->shared_from_this());
+        try
+        {
+            _raw_sensor->open(resolved_req);
+        }
+        catch (const std::runtime_error& e)
+        {
+            // Throw a more informative exception
+            std::stringstream requests_info;
+            for (auto&& r : requests)
+            {
+                auto p = to_profile(r.get());
+                requests_info << "\tFormat: " + std::string(rs2_format_to_string(p.format)) << ", width: " << p.width << ", height: " << p.height << "\n";
+            }
+            throw recoverable_exception("Failed to resolve the request: \n" + requests_info.str()
+                , RS2_EXCEPTION_TYPE_INVALID_VALUE);
+        }
+        set_active_streams(requests);
+    }
+
+    void synthetic_sensor::close()
+    {
+        std::lock_guard<std::mutex> lock(_synthetic_configure_lock);
+        _raw_sensor->close();
+        for (auto&& entry : _profiles_to_processing_block)
+        {
+            unregister_processing_block_options(*entry.second);
+        }
+        _profiles_to_processing_block.erase(begin(_profiles_to_processing_block), end(_profiles_to_processing_block));
+        cached_requests.erase(cached_requests.begin(), cached_requests.end());
+    }
+
+    template<class T>
+    frame_callback_ptr make_callback(T callback)
+    {
+        return {
+            new internal_frame_callback<T>(callback),
+            [](rs2_frame_callback* p) { /*p->release(); */}
```

Comment:
Is there any reason to why the release is commented out?
It is causing memory leaks. #6337 . #6375 .